### PR TITLE
fix time score info on server detail

### LINKED ISSUE #6638: Times aren't sorting properly in the info tab
DDNet_EIXDi6IdsS

### PR DESCRIPTION
**Pull request by edg-l**
fixes #6638 

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/menus_browser.cpp [hunk 1]
```diff
@@ -1125,23 +1126,23 @@ void CMenus::RenderServerbrowserServerDetail(CUIRect View)
...
-					Empty = Time == 0 || Time == 9999;
+					int TempTime = abs(CurrentClient.m_Score);
+					if(TempTime != 0 && Time != 9999)
+						Time = TempTime;
 				}
 				else
 				{
-					Time = CurrentClient.m_Score;
-					Empty = Time < 0;
+					if(CurrentClient.m_Score != -9999 && absolute(CurrentClient.m_Score) >= 0)
```
Comment (heinrich5991) [inline src/game/client/components/menus_browser.cpp:1139]: The `absolute(CurrentClient.m_Score) >= 0` seems pointless. If it's supposed to check for `MIN_INT`, then the call to `absolute` is already UB.

**Comment by heinrich5991** [comment]
And what I said on Discord, I think we should fix the buggy servers instead of using `absolute` on the client side, for the explicitly specified `client_score_kind`.

File: src/game/client/components/menus_browser.cpp [hunk 2]
```diff
@@ -1125,23 +1126,23 @@ void CMenus::RenderServerbrowserServerDetail(CUIRect View)
...
+						Time = absolute(CurrentClient.m_Score);
```
[the remark this answers is not in the document]
Comment (edg-l) [reply src/game/client/components/menus_browser.cpp:1140]: we can do this on another pr

File: src/engine/client/serverbrowser.cpp [hunk 3]
```diff
@@ -563,30 +563,64 @@ void CServerBrowser::SetInfo(CServerEntry *pEntry, const CServerInfo &Info)
...
 			if(p0.m_Player && !p1.m_Player)
 				return true;
 			if(!p0.m_Player && p1.m_Player)
 				return false;
 
+			int ClientScoreKind = ScoreKind;
+			if(ClientScoreKind == CServerInfo::CLIENT_SCORE_KIND_UNSPECIFIED)
+			{
+				ClientScoreKind = CServerInfo::CLIENT_SCORE_KIND_TIME_BACKCOMPAT;
+			}
```
Comment (heinrich5991) [inline src/engine/client/serverbrowser.cpp:585]: Should probably get the same logic as in

https://github.com/ddnet/ddnet/blob/8298e677901e383806eeac27a5b94a6a2b6b882a/src/game/client/components/menus_browser.cpp#L1079-L1089

so probably needs to factored out.

File: src/game/server/teams.cpp [hunk 4]
```diff
@@ -1,5 +1,6 @@
 /* (c) Shereef Marzouk. See "licence DDRace.txt" and the readme.txt in the root of the distribution for more information. */
 #include "teams.h"
+#include "base/system.h"
```
Comment (heinrich5991) [inline src/game/server/teams.cpp:3]:
```suggestion
#include <base/system.h>
```

**Review by heinrich5991 — COMMENTED** [review]
Looks mostly good, thanks for working on this.

**Pull request merged**